tests/test_bootstrap_archive.py

```
@@ -0,0 +1,436 @@
+        layout = [locked_member("tool", payload)]
+        with tempfile.TemporaryDirectory() as temporary:
+            directory = Path(temporary) / "installed"
+            directory.mkdir(mode=0o700)
+            tool = directory / "tool"
+            tool.write_bytes(payload)
+            tool.chmod(0o755)
+            child = os.fork()
+            if child == 0:
+                try:
+                    with open_validated_executable(directory, layout, "tool") as source:
+                        with open_sealed_executable(source, layout[0]) as executable:
```

Ruff (0.16.1)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)